code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java

```diff
@@ -81,4 +85,32 @@ private String getRepositoryNameByOwner(String username, String game) throws IOE
     private GHUser getGitHubUser(String username) throws IOException {
         return gitHub.getUser(username);
     }
+
+    public String deleteReposForParticularGame(String game) throws IOException {
+        GHMyself account = gitHub.getMyself();
+        Collection<GHRepository> repos = account.getAllRepositories().values();
+        List<String> deletedRepos = new CopyOnWriteArrayList<>();
+        repos.stream().parallel()
+                .filter(repo->repo.getName().startsWith(REPO_PREFIX) && repo.getName().endsWith("-"+game))
+                .forEach(repo-> {
+                    try {
+                        repo.delete();
+                        deletedRepos.add(repo.getName());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+        return generateResult(deletedRepos);
+    }
+
+    private String generateResult(List<String> deletedRepos) {
```

Comment:
GitManager is now responsible for more than one thing which is breaking the single responsibility principle;
generating a result should be done by a separate class